### PR TITLE
Extract `infer_independent` implementation into a function

### LINKED ISSUE #59: Extract `infer_independent` implementation into a function
All [this](https://github.com/JuBiotech/calibr8/blob/v6.6.0/calibr8/core.py#L730-L795) code uses only `self.likelihood`, so it can be easily extracted into a function.

This makes it a little easier to test, or to re-use in other contexts.

### PR DESCRIPTION
**Pull request by michaelosthege**
Closes #59

### CODE REVIEW

File: calibr8/core.py [hunk 1]
```diff
@@ -727,71 +826,13 @@ def infer_independent(
...
-        return ContinuousUnivariateInference(
-            median,
-            eti_x=eti_x,
-            eti_pdf=eti_pdf,
-            eti_prob=eti_prob,
-            hdi_x=hdi_x,
-            hdi_pdf=hdi_pdf,
-            hdi_prob=hdi_prob,
+        return infer_independent(
+            likelihood=self.likelihood,
```
Comment (lhelleckes) [inline calibr8/core.py:830]: although it makes no difference, I would prefer the order as specified above (although you use kwargs only)

**Review by lhelleckes — APPROVED** [review]
Just a minor detail that you may also decide to ignore.

File: calibr8/core.py (hunk 1 above)
Comment (michaelosthege) [reply calibr8/core.py:830]: I don't understand, this is the same order as in both the method and the function signatures?

Comment (lhelleckes) [reply calibr8/core.py:830]: no, y comes before likelihood in the function above? 
y is positional and the rest kw-based, but here you insert likelihood before y. 

No change in the end, I just don't find it very pretty :)

File: calibr8/core.py [hunk 2]
```diff
@@ -689,6 +689,105 @@ def load(cls, filepath: os.PathLike):
         return obj
 
 
+def infer_independent(
+    y: Union[int, float, numpy.ndarray],
+    *,
```
Comment (michaelosthege) [inline calibr8/core.py:694]:
```suggestion
    *,
    y: Union[int, float, numpy.ndarray],
```

File: calibr8/core.py [hunk 3]
```diff
@@ -689,6 +689,105 @@ def load(cls, filepath: os.PathLike):
...
+    likelihood: Callable,
```
Comment (michaelosthege) [inline calibr8/core.py:695]:
```suggestion
    *,
    likelihood: Callable,
    y: Union[int, float, numpy.ndarray],
```

Comment (lhelleckes) [reply calibr8/core.py]: Thx, safer this way I guess.

**Pull request merged**